src/messages/MOSDRepOp.h

```diff
@@ -125,12 +170,15 @@ class MOSDRepOp : public Message {
 public:
   const char *get_type_name() const { return "osd_repop"; }
   void print(ostream& out) const {
-    out << "osd_repop(" << reqid
-	<< " " << pgid
-	<< " " << poid;
-    out << " v " << version;
-    if (updated_hit_set_history)
-      out << ", has_updated_hit_set_history";
+    out << "osd_repop(";
+    if (!partial_decode_needed) {
+      out << reqid
+          << " " << pgid
+          << " " << poid;
+      out << " v " << version;
+      if (updated_hit_set_history)
+        out << ", has_updated_hit_set_history";
+    }
```

Comment:
The reqid and pgid are always present (assert !partial_decode_needed).  the rest (poid, version, hit_set_history are condition on final_decode_needed)